crates/common/mqtt_client/src/lib.rs

```diff
@@ -207,8 +213,26 @@ impl Client {
             mqtt_options.set_max_packet_size(packet_size, packet_size);
         }
 
-        let (mqtt_client, eventloop) =
+        let (mqtt_client, mut eventloop) =
             rumqttc::AsyncClient::new(mqtt_options, config.queue_capacity);
+
+        loop {
+            match eventloop.poll().await {
+                Ok(Event::Incoming(Packet::ConnAck(ack)))
+                    if ack.code == ConnectReturnCode::Success =>
+                {
+                    break
+                }
```

Comment:
What is not clear to me here is: "does the underlying `rumqttc` client try to reconnect if the connection has been rejected".

* I'm afraid this would not be the case. And it would make sense: What is the point to retry if the connection has been rejected because of missing credentials?
* For that reason, it's better to stop here with an error.